File: RemoveRecursionRefactoring/src/ro/pub/cs/diploma/ir/Visitor.java
package ro.pub.cs.diploma.ir;

import org.jetbrains.annotations.NotNull;

public interface Visitor {
  void visit(@NotNull final Block block);

  void visit(@NotNull final ConditionalJumpStatement conditionalJumpStatement);

  void visit(@NotNull final NormalStatement normalStatement);

  void visit(@NotNull final ReturnStatement returnStatement);

  void visit(@NotNull final UnconditionalJumpStatement unconditionalJumpStatement);

  void visit(@NotNull final SwitchStatement switchStatement);
}

File: inspection/src/main/kotlin/ro/pub/cs/diploma/ir/Visitor.kt
package ro.pub.cs.diploma.ir

interface Visitor {
    fun visit(block: Block)

    fun visit(conditionalJumpStatement: ConditionalJumpStatement)

    fun visit(normalStatement: NormalStatement)

    fun visit(returnStatement: ReturnStatement)

    fun visit(unconditionalJumpStatement: UnconditionalJumpStatement)

    fun visit(switchStatement: SwitchStatement)
}
